roachtest: allow non-interactive roachstress.sh

tbg opened the pull request:
This allows using `roachstress.sh` for fully autonomous `git bisect
runs`, such as

```
git bisect run /bin/bash -c \
  'TEST=splits/load/uniform/nodes=3 LOCAL=y COUNT=1 SHORT=y ./pkg/cmd/roachtest/roachstress.sh'
````

which was helpful in
https://github.com/cockroachdb/cockroach/issues/69411.

Release justification: testing only change
Release note: None

tbg committed:
roachtest: allow non-interactive roachstress.sh

This allows using `roachstress.sh` for fully autonomous `git bisect
runs`, such as

```
git bisect run /bin/bash -c \
  'TEST=splits/load/uniform/nodes=3 LOCAL=y COUNT=1 SHORT=y ./pkg/cmd/roachtest/roachstress.sh'
````

which was helpful in
https://github.com/cockroachdb/cockroach/issues/69411.

Release justification: testing only change
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69532)
<!-- Reviewable:end -->

tbg committed:
roachtest: fix buglet in roachstress.sh

If we switch between local and non-local mode, the workload binary may
need to be rebuilt (unlike the roachtest binary, which is "always
local"). We failed to do this properly since the caching was triggered
by the existence of the `roachtest` binary.

Release justification: testing-only change
Release note: None

erikgrinaker reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69532#-:-MiMC_RL3hTF9x3pn1eK:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/cmd/roachtest/roachstress.sh, line 15 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/69532#-MiMC56g0VR6P30it9zJ:-MiMC56g0VR6P30it9zK:b-ji7ohv) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e3a9a6258576f17798ed638367dc5cd39f77a94/pkg/cmd/roachtest/roachstress.sh#L15)):*
> ```Shell
> 
> # Read user input.
> if [ -z "${TEST-}" ]; then read -r -e -p "Test regexp: " TEST; fi
> ```

nit: this can also use `-v TEST`


<!-- Sent from Reviewable.io -->

tbg committed:
roachtest: use ! -v instead of -z in roachstress.sh

As suggested by Erik.

Release justification: testing-only change
Release note: None

tbg commented:
bors r=erikgrinaker
